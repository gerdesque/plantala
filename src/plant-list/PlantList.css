.image-item {
  padding: 2em 0;
  filter: brightness(.9);
}

.react-multi-carousel-item--active {
  padding: 1em 0;
  filter: brightness(1);
  transform: perspective(800px) translate3d(0, 0, 25px);
}
