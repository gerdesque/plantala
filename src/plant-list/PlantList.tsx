import React from 'react';
import { Button, Card, CardActions, CardMedia, CardContent, Grid, Typography } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import { IPlant } from '../plant/Plant';

interface IPlantListProps {
  plants: IPlant[]
}

const useStyles = makeStyles((theme) => ({
  plant: {
    height: '100%',
    display: 'flex',
    flexDirection: 'column',
  },
  plantMedia: {
    paddingTop: '100%',
  },
  plantContent: {
    flexGrow: 1,
    backgroundColor: theme.palette.secondary.dark,
  },
}));

export default function PlantList({ plants }: IPlantListProps) {
  const classes = useStyles();

  return (
    <Grid container spacing={4} justify="center">
      {plants.map((plant) => (
        <Grid item key={plant.name} xs={12} sm={6} md={4}>
          <Card className={classes.plant}>
            <CardMedia
              className={classes.plantMedia}
              image={`${process.env.PUBLIC_URL}/assets/${plant.source}`}
              title={plant.name}
            />
            <CardContent className={classes.plantContent}>
              <Typography gutterBottom variant="h5" component="h2">
                {plant.name}
              </Typography>
              <Typography>
                {plant.description}
              </Typography>
            </CardContent>
            <CardActions>
              <Button size="small" color="secondary">
                Mehr erfahren
              </Button>
            </CardActions>
          </Card>
        </Grid>
      ))}
    </Grid>
  );
}
