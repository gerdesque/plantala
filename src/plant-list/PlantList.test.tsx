import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import PlantList from './PlantList';
import { Card, Grid } from '@material-ui/core';

describe('PlantList', () => {

  let shallow: any;
  let wrapper: any;

  beforeEach(() => {
    shallow = createShallow();
    wrapper = shallow(<PlantList plants={[]} />);
  });

  it('should render correctly', () => expect(wrapper).toMatchSnapshot());

  it('should render a <Grid /> container', () => {
    expect(wrapper.find(Grid)).toHaveLength(1);
  });

  it('should render no <Card /> component', () => {
    expect(wrapper.find(Card)).toHaveLength(0);
  });

  it('should render no 3 <Card /> components', () => {
    const plants = [
      {name: 'Lathraea squamaria', source: '1.png', description: 'Lehrtafel von Carl von Tubeuf'},
      {name: 'Oedogonium diplandrum Juranyi', source: '2.png'},
      {name: 'Nicotiana Tabacum L.', source: '3.png'}
    ]
    wrapper = shallow(<PlantList plants={plants} />);
    expect(wrapper.find(Card)).toHaveLength(3);
  });
  

});
