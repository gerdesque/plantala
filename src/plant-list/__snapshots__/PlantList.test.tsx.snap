// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlantList should render correctly 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  justify="center"
  spacing={4}
/>
`;
